### PR TITLE
[CI][TESTS][Navi21] Basic support for Navi21

### PR DESCRIPTION
**Pull request by atamazov**
Partially (FP32 for OCL backend) implements issue #1022.

- Added Smoke and Full FP32 testing stages for Navi21 with **_OpenCL_** backend.
  - Why not HIP backend: HIP compiler as of 4.2 does not support gfx1030 by default, as well as rocBLAS etc. Therefore we need to wait for 4.3 release before enabling gfx1030 testing with HIP backend. Symptom: `"hipErrorNoBinaryForGpu: Unable to find code object for all current devices!"` More info about this problem is available [at this private link](http://gerrit-git.amd.com/c/compute/ec/hip/+/519434/5/docs/markdown/hip_faq.md). However, the most of the Navi21 functionality can be tested with OpenCL from 4.2.
- Added W/A for issue #1053.

### CODE REVIEW

**Comment by atamazov** [comment]
Ready for review!

**Comment by atamazov** [comment]
Please review.

**Comment by atamazov** [comment]
@averinevg Please review as well, to learn how it works etc.

File: Jenkinsfile [hunk 1]
```diff
@@ -346,6 +346,12 @@ pipeline {
                         buildHipClangJobAndReboot(build_type: 'debug', config_targets: Smoke_targets)
                     }
                 }
+                stage('Fp32 OpenCL Debug gfx1030') {
+                    agent{ label rocmnode("navi21") }
+                    steps{
+                        buildHipClangJobAndReboot(compiler: 'g++', build_type: 'debug', config_targets: Smoke_targets, build_env: extra_log_env, gpu_arch: "gfx1030")
```
Comment (averinevg) [inline Jenkinsfile:352]: build**HipClang**JobAndReboot() is a little bit distracting here

File: Jenkinsfile [hunk 2]
```diff
@@ -581,6 +587,12 @@ pipeline {
                         buildHipClangJobAndReboot( setup_flags: Full_test, build_env: WORKAROUND_iGemm_936)
                     }
                 }
+                stage('Fp32 OpenCL All gfx1030') {
+                    agent{ label rocmnode("navi21") }
+                    steps{
+                        buildHipClangJobAndReboot(compiler: 'g++', build_env: extra_log_env, gpu_arch: "gfx1030")
```
Comment (averinevg) [inline Jenkinsfile:593]: Here is the same as in line 352

File: Jenkinsfile (hunk 1 above)
Comment (atamazov) [reply Jenkinsfile:352]: [Quality] Absolutely :face_with_head_bandage:. It worth refactoring this legacy stuff, but this is out of the scope of this PR.

/cc @shurale-nkn

File: Jenkinsfile (hunk 2 above)
Comment (atamazov) [reply Jenkinsfile]: Thanks for this finding! Fixed.

**Review by DrizztDoUrden — APPROVED** [review]

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -1104,7 +1122,8 @@ else()
     )
 endif() #if CODECOV_TEST
 
-add_custom_test(test_conv_igemm_dynamic SKIP_UNLESS_ALL
+if(NOT (MIOPEN_TEST_GFX1030 AND WORKAROUND_ISSUE_1054))
+add_custom_test(test_conv_igemm_dynamic SKIP_UNLESS_ALL GFX1030_ENABLED
```
Comment (carlushuang) [inline test/CMakeLists.txt:1126]: @atamazov this test `test_conv_igemm_dynamic` is only for vega. and here should be `GFX1030_DISABLED`. In this case the `WORKAROUND_ISSUE_1054` can be omitted

Comment (atamazov) [reply test/CMakeLists.txt:1126]: Fixed, thanks.

**Comment by atamazov** [comment]
Please re-review.

File: test/CMakeLists.txt (hunk 3 above)
Comment (carlushuang) [reply test/CMakeLists.txt:1126]: @atamazov should be `GFX1030_DISABLED` or just omit?

Comment (carlushuang) [reply test/CMakeLists.txt:1126]: @atamazov should be `GFX1030_DISABLED` or just ommited?

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -394,7 +408,7 @@ endfunction()
 # 2)Second options type describes support GPU types (gfx900, gfx906, gfx908 ...)
 #   The option can be enabled or disabled by using '_ENABLED' and '_DISABLED' suffix.
 #   If nothing is specified, the default value is taken.
-#       Default: VEGA=enabled, GFX908=disabled.
+#       Default: VEGA=enabled, all others disabled.
```
Comment (atamazov) [inline test/CMakeLists.txt:411]: :anchor:

File: test/CMakeLists.txt (hunk 3 above)
Comment (atamazov) [reply test/CMakeLists.txt:1126]: @carlushuang Omitting is enough, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1038/files#r677834654

Comment (atamazov) [reply test/CMakeLists.txt]: @carlushuang Now I see)) Previous commit was incorrect. Fixed.

**Comment by atamazov** [comment]
Now should be Ok, please re-review.

File: test/CMakeLists.txt (hunk 3 above)
Comment (carlushuang) [reply test/CMakeLists.txt]: @atamazov ok

**Review by carlushuang — APPROVED** [review]
LGTM

**Pull request merged**